src/util/getSubMatrix.js

```diff
@@ -0,0 +1,38 @@
+import { xFindClosestIndex } from 'ml-spectra-processing';
+
+/**
+ * Return the submatrix, times, and mass x axis for each range
+ * @param {Object} range - from - to of the TIC
+ * @return {Array<Object>} - List with submatrix, times and mass x axis
+ */
+export function getSubMatrix(chromatogram, range) {
+  let { from, to } = range;
+  let fromIndex = chromatogram.getClosestTime(from);
+  let toIndex = chromatogram.getClosestTime(to);
+
+  let data = chromatogram.series.ms.data.slice(fromIndex, toIndex);
+  let times = chromatogram.times.slice(fromIndex, toIndex);
+
+  let xAxis = new Set();
+  for (let i = 0; i < data.length; i++) {
+    let spectrum = data[i];
+    for (let j = 0; j < spectrum[0].length; j++) {
+      xAxis.add(Math.round(spectrum[0][j]));
+    }
+  }
+  xAxis = Array.from(xAxis).sort((a, b) => a - b);
+  const nbPoints = xAxis.length;
+  const matrix = new Array(data.length);
+  for (let i = 0; i < data.length; i++) {
+    let element = data[i];
+    let row = new Float32Array(nbPoints);
```

Comment:
why are you using a Float32Array here?

Reply:
there is not a strong razon, I did it for the zeros initialization. Do you have a suggestion?